python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition.py

```diff
@@ -436,86 +458,66 @@ def asset_key(self) -> AssetKey:
+
+    @cached_property
+    def node_cursor(self) -> Optional[AutomationConditionNodeCursor]:
+        """Cursor value storing information about this specific evaluation node, if required."""
+        if not self.condition.requires_cursor:
+            return None
+        return AutomationConditionNodeCursor(
+            true_subset=self.true_subset,
+            candidate_subset=get_serializable_candidate_subset(
+                self.context.candidate_slice.convert_to_valid_asset_subset()
             ),
+            subsets_with_metadata=self._subsets_with_metadata,
+            extra_state=self._extra_state,
         )
 
-    @staticmethod
-    def create_from_children(
-        context: "AutomationContext",
-        true_slice: AssetSlice,
-        child_results: Sequence["AutomationResult"],
-        extra_state: Optional[Union[AssetSubset, Sequence[AssetSubset]]] = None,
-    ) -> "AutomationResult":
-        """Returns a new AutomationResult from the given child results."""
-        return AutomationResult._create(
-            context=context,
-            true_slice=true_slice,
-            subsets_with_metadata=[],
-            extra_state=extra_state,
-            child_results=child_results,
+    @cached_property
+    def _serializable_evaluation(self) -> AutomationConditionEvaluation:
+        return AutomationConditionEvaluation(
+            condition_snapshot=self.condition.get_snapshot(self.condition_unique_id),
+            true_subset=self.true_subset,
+            candidate_subset=get_serializable_candidate_subset(
+                self.context.candidate_slice.convert_to_valid_asset_subset()
+            ),
+            subsets_with_metadata=self._subsets_with_metadata,
+            start_timestamp=self.start_timestamp,
+            end_timestamp=self.end_timestamp,
+            child_evaluations=[
+                child_result.serializable_evaluation for child_result in self.child_results
+            ],
         )
 
-    @staticmethod
-    def create(
-        context: "AutomationContext",
-        true_slice: AssetSlice,
-        subsets_with_metadata: Sequence[AssetSubsetWithMetadata] = [],
-        extra_state: Optional[Union[AssetSubset, Sequence[AssetSubset]]] = None,
-    ) -> "AutomationResult":
-        """Returns a new AutomationResult from the given parameters."""
-        return AutomationResult._create(
-            context=context,
-            true_slice=true_slice,
-            subsets_with_metadata=subsets_with_metadata,
-            extra_state=extra_state,
-            child_results=[],
-        )
+    @property
+    def serializable_evaluation(self) -> AutomationConditionEvaluation:
+        """Serializable representation of the evaluation of this condition."""
+        return self._serializable_evaluation_override or self._serializable_evaluation
+
+    def update_serializable_evaluation(self, evaluation: AutomationConditionEvaluation) -> None:
```

Comment:
what is going on here? Very scary to have mutations in class that has `cached_property`s